### ISSUE TITLE
Failed to push multiple apps from single manifest

### ISSUE THREAD

**Issue by pivotal-studionow** [cloudfoundry/cli#818]
Today we started seeing our deploy failing on CircleCI where the cli is downloaded for each run. We then upgraded the cli on our OSX dev station and saw the same error. Note that the first app in our manifest, watson-web, deploys just fine. The CLI fails when the second app in the manifest tries to deploy. For reference, cf version 6.16.1+924508c-2016-02-26 works as expected.

### Command

`cf push -f manifest.yml`

### CLI Version

6.17.0+5d0be0a-2016-04-15

### Error

Oh no. Something's gone wrong. I've tried to roll back but you should check to see if everything is OK.: Error executing cli core command

### Stack Trace

```
App watson-worker is a worker, skipping route creation
Uploading watson-worker...
Uploading app files from: /Users/pivotal/workspace/watson
Uploading 15.5M, 4202 files
Done uploading
OK
Binding service papertrail to app watson-worker in org studionow / space development as pivotal-studionow@googlegroups.com...
OK

Stopping app watson-worker in org studionow / space development as pivotal-studionow@googlegroups.com...
OK

Starting app watson-worker in org studionow / space development as pivotal-studionow@googlegroups.com...
panic: send on closed channel

goroutine 894 [running]:
panic(0x6ea700, 0xc82065a660)
	/usr/local/Cellar/go/1.6.1/libexec/src/runtime/panic.go:464 +0x3e6
github.com/cloudfoundry/cli/cf/api.(*LoggregatorLogsRepository).flushMessageQueue.func1(0xc8213b2000)
	/Users/cffadmin/concourse-work-dir/containers/00002kcgp1v/tmp/build/254d4083/gopath/src/github.com/cloudfoundry/cli/cf/api/log_repository.go:107 +0x40
github.com/cloudfoundry/cli/cf/api.(*Loggregator_SortedMessageQueue).EnumerateAndClear(0xc8200fe600, 0xc820401748)
	/Users/cffadmin/concourse-work-dir/containers/00002kcgp1v/tmp/build/254d4083/gopath/src/github.com/cloudfoundry/cli/cf/api/loggregator_message_queue.go:46 +0x103
github.com/cloudfoundry/cli/cf/api.(*LoggregatorLogsRepository).flushMessageQueue(0xc8204244c0, 0xc8212cd200)
	/Users/cffadmin/concourse-work-dir/containers/00002kcgp1v/tmp/build/254d4083/gopath/src/github.com/cloudfoundry/cli/cf/api/log_repository.go:108 +0x45
github.com/cloudfoundry/cli/cf/api.(*LoggregatorLogsRepository).TailLogsFor.func1(0xc8202e8e80, 0xc8204244c0, 0xc8212cd200)
	/Users/cffadmin/concourse-work-dir/containers/00002kcgp1v/tmp/build/254d4083/gopath/src/github.com/cloudfoundry/cli/cf/api/log_repository.go:89 +0x78
created by github.com/cloudfoundry/cli/cf/api.(*LoggregatorLogsRepository).TailLogsFor
	/Users/cffadmin/concourse-work-dir/containers/00002kcgp1v/tmp/build/254d4083/gopath/src/github.com/cloudfoundry/cli/cf/api/log_repository.go:91 +0x383
```
### Platform Details

Mac OS X 10.11 & CircleCI

### Shell

iTerm & CircleCI

**Comment by pivotal-studionow** [comment]
This is a Python-Flask app
Here's our redacted manifest.yml:
```yml
---
applications:
- name: watson-web
  instances: 4
  memory: 128M
  buildpack: https://github.com/cloudfoundry/python-buildpack
  command: newrelic-admin run-program python watson/routes.py
  env:
    WATSON_ENV: devpws
    WATSON_HOME: /home/vcap/app
    API_URL: https://watson-web.cfapps.io/api
    HOST_URL: https://watson-web.cfapps.io
    NEW_RELIC_CONFIG_FILE: newrelic-pws.ini
    NEW_RELIC_ENVIRONMENT: devpws
  services:
    - papertrail
- name: watson-worker
  instances: 2
  memory: 128M
  buildpack: https://github.com/cloudfoundry/python-buildpack
  command: watson/sqs_worker.py --interactive start
  no-route: true
  health-check-type: process
  env:
    WATSON_ENV: devpws
    WATSON_HOME: /home/vcap/app
    API_URL: https://watson-web.cfapps.io/api
    HOST_URL: https://watson-web.cfapps.io
  services:
    - papertrail

```

**Comment by pivotal-studionow** [comment]
Updated original post, the plugin is not needed to reproduce this error

**Comment by dkoper** [comment]
This looks like an issue we found during our iteration on this release and was supposed to be looked at as last thing before this release, but it looks like I mistakenly placed the story just before the next release instead. My apologies. I've moved it up so the team will get to it soon.

**Comment by tsaleh** [comment]
@pivotal-studionow It looks like you're setting `heath-check-type:` in your manifest.   [Are you sure that's supported](https://github.com/cloudfoundry/cli/issues/803)?

**Comment by pivotal-studionow** [comment]
@tsaleh not sure, but couldn't find a way to push a non-web app without it :(

**Comment by tsaleh** [comment]
@pivotal-studionow You're right - I was mistaken (#803 is now closed).  That's GREAT!

What's not great is that now I have the same problem :( :( :(

**Comment by tsaleh** [comment]
@dkoper Is there an older version I can lock to in order to get around this?

**Comment by challiwill** [comment]
@tsaleh We have a fix working it's way through our pipeline now. Hopefully we will get a release out later today or first thing Monday.

**Comment by pivotal-studionow** [comment]
@tsaleh 6.16.1 didn't have this problem for us, you can grab it off the releases page https://github.com/cloudfoundry/cli/releases/

**Comment by dkoper** [comment]
This has been fixed in cf CLI v6.17.1.
https://github.com/cloudfoundry/cli#downloads

Thanks again!
Dies Koper
CF CLI PM
